src/drivers/px4io/CMakeLists.txt

```diff
@@ -67,10 +69,11 @@ set(fw_io_bin "${PX4_BINARY_DIR}/romfs_extras/${config_io_board}.bin" CACHE FILE
 file(RELATIVE_PATH fw_io_exe_relative ${CMAKE_CURRENT_BINARY_DIR} ${fw_io_exe})
 file(RELATIVE_PATH fw_io_bin_relative ${CMAKE_CURRENT_BINARY_DIR} ${fw_io_bin})
 
-add_custom_command(OUTPUT ${fw_io_bin}
+add_custom_command(OUTPUT ${config_io_board}.bin ${fw_io_bin}
 	COMMAND ${CMAKE_COMMAND} -E make_directory ${PX4_BINARY_DIR}/romfs_extras
-	COMMAND ${CMAKE_OBJCOPY} -O binary ${fw_io_exe_relative} ${fw_io_bin_relative}
-	DEPENDS px4io_firmware
+	COMMAND ${CMAKE_OBJCOPY} -O binary ${fw_io_exe} ${config_io_board}.bin
+	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${config_io_board}.bin ${PX4_BINARY_DIR}/romfs_extras/${config_io_board}.bin
+	DEPENDS px4io_firmware px4_io-v2_default.elf
```

Comment:
I remember `add_custom_command` can't depend on a file but only on targets, so you need to add a target in-between, e.g.:

```
add_custom_target(
    px4_io_v2_default_elf_target
    DEPENDS px4_io-v2_default.elf
    )
```

See: https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/